mesonbuild/mesonmain.py

```diff
@@ -155,6 +155,23 @@ def run(original_args, mainfile):
         print('Please update your environment')
         return 1
 
+    # These are for Python 3.5 & 3.6 as meson is not tested on non-UTF-8 locales
```

Comment:
If it works out of the box with python3.7, I think we should make this hack conditional on python < 3.7.

Reply:
The test will always be false on 3.7, should we add an extra test for the version just in case?

Reply:
Actually that's a lie, re-reding PEP 540. It will only be UTF-8 if the locale is POSIX or a UTF-8 one, so we actually need to this on newer Python too on those platforms I think.

Reply:
Please update the comment then.

Reply:
and link to PEP probably.

Reply:
Comment updated